Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2318,8 +2318,8 @@ private void processNextQueueItem() {
                             networkCount_ = 0;
                             handleFailure(requestQueue_.getSize() - 1, BranchError.ERR_NO_SESSION);
                         }
-                        //All request except open and install need a session to execute
-                        else if (!(req instanceof ServerRequestInitSession) && (!hasSession() || !hasDeviceFingerPrint())) {
+                        // Determine if a session is needed to execute (SDK-271)
+                        else if (requestNeedsSession(req) && !isSessionAvailableForRequest()) {
```

Comment:
This looks good. But is this reachable? In my case `if`-condition with check on `instance of ServerRequestRegisterInstall ` in [line 2316](https://github.com/BranchMetrics/android-branch-deep-linking-attribution/pull/700/files#diff-11fc8f6026de99b60216e9aca37d6d26R2316) catch the request and fail with not initialized session.

Reply:
Are you dropping into the block from the check on line 2316 (that was unchanged in this PR) or are you dropping into the block from check on line 2322?

Reply:
I'm dropping on unchanged lines on 2316. 
This line checks if request from type `ServerRequestRegisterInstall`. On link generation this is not and without tracking I haven't a user. Or did I miss something?